test/integration/requiredOptional.spec.ts

```diff
@@ -0,0 +1,160 @@
+      assert.equal(result._response.status, 200);
+    });
+
+    it("should handle postOptionalStringParameter", async () => {
+      const result = await client.explicit.postOptionalStringParameter();
+      assert.equal(result._response.status, 200);
+    });
+
+    it("should handle postOptionalStringProperty", async () => {
+      const result = await client.explicit.postOptionalStringProperty();
+      assert.equal(result._response.status, 200);
+    });
+
+    it("should handle postRequiredArrayHeader", async () => {
+      try {
+        await client.explicit.postRequiredArrayHeader(null as any);
+        assert.fail("Expected client to throw");
+      } catch (error) {
+        assert(
+          error.message.indexOf("cannot be null or undefined") !== -1,
+          "Expected error to contain 'cannot be null or undefined'"
+        );
+      }
+    });
+
+    it("should handle postRequiredArrayParameter", async () => {
+      try {
+        await client.explicit.postRequiredArrayParameter(null as any);
+        assert.fail("Expected client to throw");
+      } catch (error) {
+        assert(
+          error.message.indexOf("cannot be null or undefined") !== -1,
+          "Expected error to contain 'cannot be null or undefined'"
+        );
+      }
+    });
+
+    it("should handle postRequiredIntegerHeader", async () => {
+      try {
+        await client.explicit.postRequiredIntegerHeader(null as any);
+        assert.fail("Expected client to throw");
+      } catch (error) {
+        assert(
+          error.message.indexOf("cannot be null or undefined") !== -1,
+          "Expected error to contain 'cannot be null or undefined'"
+        );
+      }
+    });
+
+    it("should handle postRequiredIntegerParameter", async () => {
+      try {
+        await client.explicit.postRequiredIntegerParameter(null as any);
+        assert.fail("Expected client to throw");
+      } catch (error) {
+        assert(
+          error.message.indexOf("cannot be null or undefined") !== -1,
+          "Expected error to contain 'cannot be null or undefined'"
+        );
+      }
+    });
+
+    it("should handle postRequiredStringHeader", async () => {
+      try {
+        await client.explicit.postRequiredStringHeader(null as any);
+        assert.fail("Expected client to throw");
+      } catch (error) {
+        assert(
+          error.message.indexOf("cannot be null or undefined") !== -1,
+          "Expected error to contain 'cannot be null or undefined'"
+        );
+      }
+    });
+
+    it("should handle postRequiredStringParameter", async () => {
+      try {
+        await client.explicit.postRequiredStringParameter(null as any);
+        assert.fail("Expected client to throw");
+      } catch (error) {
+        assert(
+          error.message.indexOf("cannot be null or undefined") !== -1,
+          "Expected error to contain 'cannot be null or undefined'"
+        );
+      }
+    });
+  });
+});
```

Comment:
There are a few extra ones in Python's, e.g.: https://github.com/Azure/autorest.python/blob/4cfd3e8e15f902e76c9ae9145ced542a459e78ef/test/vanilla/AcceptanceTests/test_required_optional.py#L92. Can we add them as well?